app/src/main/java/com/sun/makeupwindow/data/source/local/dao/ProductDAOImpl.kt

```diff
@@ -0,0 +1,71 @@
+package com.sun.makeupwindow.data.source.local.dao
+
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.data.source.local.db.AppDatabase
+
+class ProductDAOImpl private constructor(
+    appDatabase: AppDatabase
+) : ProductDAO {
+
+    private val appDatabase = appDatabase
```

Comment:
Như anh Thọ đã suggest em thì em có thử xem thêm val vào có gì khác ? 

```suggestion
class ProductDAOImpl private constructor(
    private val appDatabase: AppDatabase
) : ProductDAO {

    private val appDatabase = appDatabase
```

Reply:
đúng là có khác anh ạ